src/cpp/src/lm_encoding.cpp

```diff
@@ -105,6 +105,20 @@ std::pair<EncodedResults, int32_t> get_lm_encoded_results(
 
     auto logits = m_llm.get_tensor("logits");
 
+    // if slice matmul is not applied logits will contains not only result tokens
+    size_t vocab_size = logits.get_shape().back();
+    if (!m_embedding.has_value()) {
+        ov::Tensor new_logits = ov::Tensor(logits.get_element_type(), {batch_size, 1, vocab_size});
```

Comment:
I suppose sliced logits already have `batch_size, 1, vocab_size` size ?

see https://github.com/openvinotoolkit/openvino.genai/pull/1215#discussion_r1844321285

Reply:
yes, updated according to way with update_processed_tokens_num, thank you !